src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Rss20FeedFormatter.cs

```diff
@@ -1199,6 +1181,65 @@ private bool checkTextInput(SyndicationTextInput textInput)
             }
         }
 
+        private async Task ReadOptionalTasgAsync(XmlBuffer buffer, SyndicationFeed result)
+        {
+            XmlReader reader = XmlReaderWrapper.CreateFromReader(buffer.GetReader(0));
+            try
```

Comment:
The reader returned by the buffer shouldn't be wrapped as it's an XmlDictionaryReader reading from an in-memory stream/buffer so will never go async. You should call the sync methods on the returned buffer.